client/client.py

```diff
@@ -0,0 +1,126 @@
+import requests
+from datetime import datetime
+from dataclasses import dataclass
+
+
+@dataclass
+class Status:
+    """Represents the status of a file upload and processing."""
+    status: str
+    filename: str
+    timestamp: datetime
+    explanation: str
+
+    def is_done(self) -> bool:
+        """Check if the status is 'done'."""
+        return self.status == 'done'
+
+
+class WebAppClient:
+    """Client for interacting with the web application."""
+
+    def __init__(self, base_url: str):
+        self.base_url = base_url
+
+    def upload(self, file_path: str) -> str:
+        """
+        Upload a file to the web application.
+        Args:
+            file_path (str): The path to the file to upload.
+        Returns:
+            str: The unique identifier (UID) associated with the uploaded file.
+        Raises:
+            Exception: If the upload fails with a non-200 status code.
+        """
+        url = f"{self.base_url}/upload"
+        with open(file_path, 'rb') as file:
+            response = requests.post(url, files={'file': file})
+
+        if response.status_code == 200:
+            json_data = response.json()
+            return json_data['uid']
+        else:
+            raise Exception(f"Upload failed with status code {response.status_code}")
+
+    def status(self, uid: str) -> type[Status]:
```

Comment:
Separate code to small functions.  Read more: https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be